editor/animation_track_editor_plugins.cpp

```diff
@@ -101,47 +101,63 @@ void AnimationTrackEditColor::draw_key_link(int p_index, float p_pixels_sec, int
+	x_to = MIN(x_to, p_clip_right);
+
+	int y_from = (get_size().height - fh) / 2;
 
 	if (x_from > p_clip_right || x_to < p_clip_left) {
 		return;
 	}
 
-	Color color = get_animation()->track_get_key_value(get_track(), p_index);
-	Color color_next = get_animation()->track_get_key_value(get_track(), p_index + 1);
+	Vector<Color> color_samples;
+	color_samples.append(get_animation()->track_get_key_value(get_track(), p_index));
 
-	if (x_from < p_clip_left) {
-		float c = float(p_clip_left - x_from) / (x_to - x_from);
-		color = color.lerp(color_next, c);
-		x_from = p_clip_left;
-	}
+	if (get_animation()->track_get_type(get_track()) == Animation::TYPE_VALUE) {
+		if (get_animation()->track_get_interpolation_type(get_track()) != Animation::INTERPOLATION_NEAREST &&
+				(get_animation()->value_track_get_update_mode(get_track()) == Animation::UPDATE_CONTINUOUS ||
+						get_animation()->value_track_get_update_mode(get_track()) == Animation::UPDATE_CAPTURE) &&
+				!Math::is_zero_approx(get_animation()->track_get_key_transition(get_track(), p_index))) {
+			float start_time = get_animation()->track_get_key_time(get_track(), p_index);
+			float end_time = get_animation()->track_get_key_time(get_track(), p_index + 1);
 
-	if (x_to > p_clip_right) {
-		float c = float(p_clip_right - x_from) / (x_to - x_from);
-		color_next = color.lerp(color_next, c);
-		x_to = p_clip_right;
-	}
+			Color color_next = get_animation()->value_track_interpolate(get_track(), end_time);
 
-	int y_from = (get_size().height - fh) / 2;
+			if (!color_samples[0].is_equal_approx(color_next)) {
+				color_samples.resize(1 + (x_to - x_from) / 64); // Make a color sample every 64 px.
+				for (int i = 1; i < color_samples.size(); i++) {
+					float j = i;
+					color_samples.write[i] = get_animation()->value_track_interpolate(
+							get_track(),
+							Math::lerp(start_time, end_time, j / color_samples.size()));
+				}
+			}
+			color_samples.append(color_next);
+		} else {
+			color_samples.append(color_samples[0]);
+		}
+	} else {
+		color_samples.append(get_animation()->track_get_key_value(get_track(), p_index + 1));
+	}
 
 	Vector<Vector2> points;
 	Vector<Color> colors;
+	points.resize(color_samples.size() * 2);
+	colors.resize(color_samples.size() * 2);
 
-	points.push_back(Vector2(x_from, y_from));
-	colors.push_back(color);
+	for (int i = 0; i < (color_samples.size()); i++) {
+		int x_pos = Math::lerp(x_from, x_to, float(i) / (color_samples.size() - 1));
 
-	points.push_back(Vector2(x_to, y_from));
-	colors.push_back(color_next);
-
-	points.push_back(Vector2(x_to, y_from + fh));
-	colors.push_back(color_next);
-
-	points.push_back(Vector2(x_from, y_from + fh));
-	colors.push_back(color);
-
-	draw_primitive(points, colors, Vector<Vector2>());
+		points.write[i] = Vector2(x_pos, y_from);
+		colors.write[i] = color_samples[i];
+		points.write[points.size() - i] = Vector2(x_pos, y_from + fh);
+		colors.write[points.size() - i] = color_samples[i];
```

Comment:
```suggestion
		points.write[points.size() - i - 1] = Vector2(x_pos, y_from + fh);
		colors.write[points.size() - i - 1] = color_samples[i];
```

Reply:
Actually something is wrong when setting easing to high value:
![image](https://user-images.githubusercontent.com/2223172/112658630-df04de00-8e53-11eb-9a72-4e45828f2f00.png)
Not sure why.

Reply:
Here's a bigger version for reference.
![image](https://user-images.githubusercontent.com/33117082/112661197-8daa1e00-8e56-11eb-9f4d-3c7e23b26e7b.png)

Reply:
After debugging a bit, it seems a bug in the renderer as it appears only when there are 8 or more points.
I fix it by splitting the polygon in chunks of 6 points.

Reply:
Uh, the code looks too complicated now .-.
But that maybe it should be fine if you add a comment saying that this is a workaround for a bug. Or alternatively you could return to the first approach...

Reply:
The same issue happens in 3.x, so I think it is better to return to the `draw_primitive` approach.

<details>
  <summary>chunked draw_polygon approach</summary>
  
  ```c++
    const int polygon_points = 3;

	Vector<Vector2> points;
	Vector<Color> colors;
	points.resize(polygon_points * 2);
	colors.resize(polygon_points * 2);

	int j = 0;
	for (int i = 0; i < color_samples.size() || j == polygon_points; i++) {
		if (j == polygon_points) {
			j = 0;
			i -= 2;
			draw_polygon(points.duplicate(), colors.duplicate());
		} else {
			int x_pos = Math::lerp(x_from, x_to, float(i) / (color_samples.size() - 1));
			points.write[j] = Vector2(x_pos, y_from);
			colors.write[j] = color_samples[i];
			points.write[points.size() - j - 1] = Vector2(x_pos, y_from + fh);
			colors.write[points.size() - j - 1] = color_samples[i];
			j++;
		}
	}

	if (j > 1) {
		Vector<Vector2> points_aux = points.subarray(points.size() - j, points.size() - 1);
		Vector<Color> colors_aux = colors.subarray(points.size() - j, points.size() - 1);
		points.resize(j);
		colors.resize(j);
		points.append_array(points_aux);
		colors.append_array(colors_aux);
		draw_polygon(points, colors);
	}
```
</details>